importer: TestFailedImportGC is flaky

@ajwerner I think https://github.com/cockroachdb/cockroach/pull/87845 is causing the flakiness.

```
--- FAIL: TestFailedImportGC (54.65s)
    test_log_scope.go:161: test logs captured to: /Users/michaelbutler/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/a340ebbb90405137386a0ed860023d6b/logTestFailedImportGC3426392160
    test_log_scope.go:79: use -show-logs to present logs inline
    sql_runner.go:101: 
        	Error Trace:	/private/var/tmp/_bazel_michaelbutler/13ba282fa2b19539d0c969c1113bb37c/sandbox/darwin-sandbox/4622/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin-fastbuild/bin/pkg/sql/importer/importer_test_/importer_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/importer/sql_runner.go:114
        	            				/private/var/tmp/_bazel_michaelbutler/13ba282fa2b19539d0c969c1113bb37c/sandbox/darwin-sandbox/4622/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin-fastbuild/bin/pkg/sql/importer/importer_test_/importer_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/importer/sql_runner.go:284
        	            				/private/var/tmp/_bazel_michaelbutler/13ba282fa2b19539d0c969c1113bb37c/sandbox/darwin-sandbox/4622/execroot/com_github_cockroachdb_cockroach/bazel-out/darwin-fastbuild/bin/pkg/sql/importer/importer_test_/importer_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/importer/import_stmt_test.go:2073
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'SCHEMA CHANGE GC' AND running_status = 'waiting for MVCC GC' AND created > '2022-09-16 22:22:22.826+00:00'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:287
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:69
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:207
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:114
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:284
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/importer.TestFailedImportGC
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/importer/import_stmt_test.go:2073
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'SCHEMA CHANGE GC' AND running_status = 'waiting for MVCC GC' AND created > '2022-09-16 22:22:22.826+00:00'': expected:
        	            	  | 1
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:522: -- test log scope end --
FAIL
I220916 22:23:08.885088 1 (gostd) testmain.go:248  [-] 1  Test //pkg/sql/importer:importer_test exited with error code 1


ERROR: exit status 1
```


Jira issue: CRDB-19671

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -2000,8 +2000,7 @@ func TestFailedImportGC(t *testing.T) {
 		ExternalIODir:            baseDir,
 		Knobs: base.TestingKnobs{
 			GCJob: &sql.GCJobTestingKnobs{
-				RunBeforeResume:      func(_ jobspb.JobID) error { <-blockGC; return nil },
-				SkipWaitingForMVCCGC: true,
+				RunBeforeResume: func(_ jobspb.JobID) error { <-blockGC; return nil },
 			},
 		},
 	}})
@@ -2027,6 +2026,9 @@ func TestFailedImportGC(t *testing.T) {
 	kvDB := tc.Server(0).DB()
 
 	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.batch_size = '10KB'`)
+	// The test assumes we'll use the MVCC range tombstone in the GC job. We need
+	// to set this cluster setting to make that true.
+	sqlDB.Exec(t, `SET CLUSTER SETTING storage.mvcc.range_tombstones.enabled = true`)
 
 	forceFailure = true
 	defer func() { forceFailure = false }()

